src/language_server.rs.in

```diff
@@ -0,0 +1,1135 @@
+
+use std::collections::HashMap;
+use serde;
+use serde::de::Error;
+use serde_json::Value;
+
+#[derive(Deserialize)]
+pub struct CancelParams {
+    /**
+     * The request id to cancel.
+     */
+    pub id: String,
+}
+
+#[derive(Deserialize)]
+pub struct DidChangeTextDocumentParams {
+    /**
+     * The document that did change. The version number points
+     * to the version after all provided content changes have
+     * been applied.
+     */
```

Comment:
I have been thinking I may do that since I suspect it could reduce the compile times a bit (by not needing to run serde on every compile). Would need to publish it to crates.io as well which I suppose is ok, a crate with just these types should be uncontroversial.

I did actually offer to publish these types in a standalone crate a while back for https://github.com/RustDT/RustLSP but he ended up doing it as well. The crate itself might be useful to pull in sometime but currently it seems a bit to experimental to be useable.